variables.tf

```diff
@@ -585,3 +478,218 @@ variable "security_group_name" {
   type        = string
   default     = "kong-security-group"
 }
+
+## ECS
+
+variable "fargate_cpu" {
+  description = "(Optional) The CPU for the Fargate Task"
+  type        = number
+  default     = 2048
+}
+
+variable "fargate_memory" {
+  description = "(Optional) The Memory for the Fargate Task"
+  type        = number
+  default     = 4096
+}
+
+variable "kong_cp_ports" {
+  description = "The ports for the Kong Data Plane"
+  type        = map(number)
+  default = {
+    "admin-api"  = 8444,
+    "admin-gui"  = 8445,
+    "status"     = 8100,
+    "clustering" = 8005,
+    "telemetry"  = 8006
+  }
+}
+
+variable "kong_dp_ports" {
+  description = "The ports for the Kong Control Plane"
+  type        = map(number)
+  default = {
+    "proxy"  = 8443,
+    "status" = 8100
+  }
+}
```

Comment:
I think these should become a single object e.g. `kong_ports` then pass in the ones you need for cp, dp, or stand alone

Reply:
What benefit would this bring? These ports are passed into the task definition which are then mapped onto the container, e.g

`    ports                       = jsonencode([for k, v in var.kong_cp_ports : v])`

I guess this would still work with a single `kong_ports` var although I'd argue it's then more complex to iterate on the ports that each role needs and not just have each container have every port open - unless of course theres an easier way

Reply:
need to have a bit of a think about the kong ports variables, I don't think the separate cp and dp ports works too well, it doesn't really accommodate a standalone role or a portal role; the default can be overridden, but then the name is very misleading. I'm thinking along the lines of a `kong_ports` variable with an empty map as a default plus a few `locals` for cp, dp, and portal ports. Then if `var.kong_ports` is empty pick the `local` value based on the `var.role`, if `var.kong_ports` isn't empty then use that instead. I'm happy to call and discuss if you prefer.

Reply:
As per our discussion, this has now been done